explore/src/main/scala/explore/targets/TargetSelectionPopup.scala

```diff
@@ -44,7 +44,7 @@ import scala.concurrent.duration._
 
 final case class TargetSelectionPopup(
   trigger:          Reuse[Button],
-  onSelected:       Target ==> Callback
+  onSelected:       (Option[Target.Id], Target) ==> Callback
```

Comment:
Nice

Reply:
Should we have case class for this? Seems to be used in a few places.

Reply:
Yeah, might be more clear. `TargetWithId` is actually used more places and is just an alias for a similar tuple. I've been wondering if that, along with `SiderealTargetWithId` and `NonsiderealTargetWithId`, should be bumped up to a case class.